File: check_lmstudio.mjs
// apps/web/check_lmstudio.mjs
// Быстрый REST-пинг LM Studio: /v1/models, non-stream chat, stream chat.
const HOST = process.env.LMSTUDIO_HOST ?? "http://192.168.0.111:1234";
const MODEL = process.env.LMSTUDIO_MODEL ?? "qwen/qwen3-14b";

const fetchJson = async (url, init) => {
    const r = await fetch(url, init);
    if (!r.ok) throw new Error(`${r.status} ${r.statusText}`);
    return await r.json();
};

async function listModels() {
    const data = await fetchJson(`${HOST}/v1/models`);
    console.log("== /v1/models ==\n", JSON.stringify(data, null, 2));
    const ids = (data.data ?? []).map(m => m.id);
    if (!ids.includes(MODEL)) {
        console.warn(`\n⚠️ Модель '${MODEL}' не найдена. Доступные:\n- ${ids.join("\n- ")}`);
    }
}

async function chatOnce() {
    const payload = {
        model: MODEL,
        messages: [
            { role: "system", content: "Отвечай кратко и по делу." },
            { role: "user", content: "Скажи привет одним предложением." }
        ],
        temperature: 0.3,
        max_tokens: 128
    };
    const data = await fetchJson(`${HOST}/v1/chat/completions`, {
        method: "POST",
        headers: { "content-type": "application/json" },
        body: JSON.stringify(payload),
    });
    console.log("\n== non-stream chat ==");
    console.log("assistant:", data.choices?.[0]?.message?.content?.trim());
    console.log("usage:", data.usage ?? "(не предоставлен сервером)");
}

async function chatStream() {
    const payload = {
        model: MODEL,
        messages: [
            { role: "system", content: "Отвечай кратко и по делу." },
            { role: "user", content: "Дай 3 факта о зелёном чае." }
        ],
        temperature: 0.3,
        max_tokens: 256,
        stream: true
    };
    const r = await fetch(`${HOST}/v1/chat/completions`, {
        method: "POST",
        headers: { "content-type": "application/json", "accept": "text/event-stream" },
        body: JSON.stringify(payload),
    });
    if (!r.ok || !r.body) throw new Error(`${r.status} ${r.statusText}`);

    console.log("\n== stream chat ==");
    const reader = r.body.getReader();
    const decoder = new TextDecoder();
    let buffer = "";
    while (true) {
        const { value, done } = await reader.read();
        if (done) break;
        buffer += decoder.decode(value, { stream: true });
        let i;
        while ((i = buffer.indexOf("\n\n")) >= 0) {
            const chunk = buffer.slice(0, i); buffer = buffer.slice(i + 2);
            if (!chunk.startsWith("data: ")) continue;
            const data = chunk.slice(6).trim();
            if (data === "[DONE]") { console.log("\n-- end of stream --"); return; }
            try {
                const j = JSON.parse(data);
                const delta = j?.choices?.[0]?.delta?.content ?? "";
                if (delta) process.stdout.write(delta);
            } catch { /* пропускаем служебные чанки */ }
        }
    }
}

(async () => {
    await listModels();
    console.log("\nℹ️ Контекстное окно через стандартный REST обычно не видно; для SDK см. getContextLength().");
    await chatOnce();
    await chatStream();
})().catch(e => { console.error("\n❌ Error:", e); process.exit(1); });

File: check_lmstudio.py
# check_lmstudio.py
# Python 3.13
import os, sys, json, asyncio, time
import httpx

HOST = os.getenv("LMSTUDIO_HOST", "http://192.168.0.111:1234")
MODEL = os.getenv("LMSTUDIO_MODEL", "qwen/qwen3-14b")  # укажи точное имя своей скачанной модели

def pretty(obj): return json.dumps(obj, ensure_ascii=False, indent=2)

async def list_models(client: httpx.AsyncClient):
    r = await client.get(f"{HOST}/v1/models", timeout=10)
    r.raise_for_status()
    data = r.json()
    print("== /v1/models ==")
    print(pretty(data))
    # Пытаемся найти наш MODEL в списке
    ids = [m.get("id") for m in (data.get("data") or [])]
    if MODEL not in ids:
        print(f"\n⚠️  Модель '{MODEL}' не найдена в списке. Доступные:\n- " + "\n- ".join(map(str, ids)))
    return data

async def chat_once(client: httpx.AsyncClient, system: str, user: str):
    payload = {
        "model": MODEL,
        "messages": [
            {"role": "system", "content": system},
            {"role": "user", "content": user},
        ],
        "temperature": 0.3,
        "max_tokens": 256
    }
    r = await client.post(f"{HOST}/v1/chat/completions", json=payload, timeout=60)
    r.raise_for_status()
    data = r.json()
    print("\n== non-stream chat ==")
    print("assistant:", data["choices"][0]["message"]["content"].strip())
    usage = data.get("usage")
    if usage:
        print("usage:", usage)
    else:
        print("usage: (не предоставлен сервером)")

async def chat_stream(client: httpx.AsyncClient, system: str, user: str):
    payload = {
        "model": MODEL,
        "messages": [
            {"role": "system", "content": system},
            {"role": "user", "content": user},
        ],
        "temperature": 0.3,
        "max_tokens": 256,
        "stream": True,
    }
    print("\n== stream chat ==")
    async with client.stream("POST", f"{HOST}/v1/chat/completions", json=payload, timeout=None) as r:
        r.raise_for_status()
        buf = []
        async for line in r.aiter_lines():
            if not line:
                continue
            if line.startswith("data: "):
                chunk = line[6:]
                if chunk.strip() == "[DONE]":
                    break
                try:
                    j = json.loads(chunk)
                    delta = j["choices"][0].get("delta", {})
                    text = delta.get("content", "")
                    if text:
                        buf.append(text)
                        sys.stdout.write(text)
                        sys.stdout.flush()
                except Exception as e:
                    # бывает служебный чанκ без delta
                    pass
        print("\n-- end of stream --")
        print("collected chars:", sum(len(x) for x in buf))

async def main():
    async with httpx.AsyncClient(headers={"Accept": "application/json"}) as client:
        await list_models(client)
        # Контекстное окно через REST обычно НЕ доступно:
        print("\nℹ️  Размер контекстного окна через стандартный REST не возвращается. "
              "Если нужен из SDK — см. Node-скрипт ниже или SDK-метод getContextLength().")
        await chat_once(client, "Отвечай кратко и по делу.", "Скажи привет одним предложением.")
        await chat_stream(client, "Отвечай кратко и по делу.", "Дай 3 факта о чёрном чае.")

if __name__ == "__main__":
    asyncio.run(main())
